lib/anony/result.rb

```diff
@@ -9,32 +9,33 @@ class Result
     OVERWRITTEN = "overwritten"
     SKIPPED = "skipped"
 
-    attr_reader :status, :fields, :error
+    attr_reader :status, :fields, :error, :record
 
     delegate :failed?, :overwritten?, :skipped?, :destroyed?, to: :status
 
-    def self.failed(error)
-      new(FAILED, error: error)
+    def self.failed(error, record = nil)
```

Comment:
Why do we allow record to be `nil` this does feel like an internal detail - could we avoid this?

Reply:
My thinking on this is that if any downstream consumers of the gem define their own strategies other than `overwrite` or `destroy`, and thus are creating their own `Result` objects, this won't be a breaking change

I could add a deprecation warning when used with a `nil` record perhaps?

Reply:
Wait, nvm. I've just re-read the docs and they clearly imply that only field-level strategies are expected to be customisable and that only `overwrite` and `destroy` top-level strategies are usable, so yeah we can just get rid of this attempt at backwards compatibility as it shouldn't be breaking :)